### PR TITLE
fix scrollToAlignment of TreeWidget to align vscode

### PR DESCRIPTION
**Pull request by shuyaqian**
Signed-off-by: shuyaqian 717749594@qq.com

#### What it does

To fix when switch files in editor , the selected node is not in the center.
theia:

vscode:

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/navigator/src/browser/navigator-widget.tsx [hunk 1]
```diff
@@ -153,7 +153,25 @@ export class FileNavigatorWidget extends FileTreeWidget {
...
+                height={this.node.offsetHeight}
+                rows={rows}
+                renderNodeRow={this.renderNodeRow}
+                scrollToAlignment={this.model.scrollAlignment}
+                scrollToRow={this.scrollToRow}
+                handleScroll={this.handleScroll}
+            />;
+        }
+        // eslint-disable-next-line no-null/no-null
+        return null;
```
Comment (vince-fugnitto) [inline packages/navigator/src/browser/navigator-widget.tsx:174]: Simply use `undefined` instead?

```suggestion
        return undefined;
```

File: packages/navigator/src/browser/navigator-model.ts [hunk 2]
```diff
@@ -43,6 +43,11 @@ export class FileNavigatorModel extends FileTreeModel {
         this.initializeRoot();
     }
 
+    protected scrollToAlignment: 'auto' | 'center';
+    get scrollAlignment(): 'auto' | 'center' {
+        return this.scrollToAlignment;
+    }
```
Comment (vince-fugnitto) [inline packages/navigator/src/browser/navigator-model.ts:49]: I'm not sure about these changes. The values `'auto'` and `'center'` are props from `TreeWidget.ViewProps` which likely should not leak to the model itself.

**Comment by shuyaqian** [comment]
I've changed this approach so that the tree node will not scroll to the center whether you click or double-click in the file tree. Scrolls only when you switch the tab.

**Comment by shuyaqian** [comment]
@vince-fugnitto Can you review this pr again?

**Comment by vince-fugnitto** [comment]
Closing due to inactivity. We can always re-open the pull-request when updates are made.

**Pull request closed without merging**